srtcore/api.cpp

```diff
@@ -1702,6 +1736,10 @@ int CUDTUnited::connectIn(CUDTSocket* s, const sockaddr_any& target_addr, int32_
        // InvertedGuard unlocks in the constructor, then locks in the
        // destructor, no matter if an exception has fired.
        InvertedLock l_unlocker (s->m_pUDT->m_bSynRecving ? &s->m_ControlLock : 0);
+
+       // record peer address
+       s->m_PeerAddr = target_addr;
+
```

Comment:
Why is this moved before `startConnect(..)`, and why was it previously after `startConnect(..)`?

Reply:
Because without this change if `startConnect` fails, this value will never be set.

Reply:
We need this value in the connection callback, while previously there was no use of it in case the connection fails, right?

Reply:
Something like that. Simply, this value wouldn't be set to any different value in case of successful connection, and if it failed, it doesn't matter for the rest of the facility anyway.